### PR TITLE
[Method References 2] LPS-165741 Convert method reference to ServiceTrackerList in module portal-template-velocity、portal-template-freemarker

### PR DESCRIPTION
**Pull request by W2jx**

### CODE REVIEW

File: modules/apps/portal-template/portal-template-freemarker/src/main/java/com/liferay/portal/template/freemarker/internal/helper/FreeMarkerTemplateContextHelper.java [hunk 1]
```diff
@@ -154,12 +152,23 @@ public void setRestrictedBeansWrapper(BeansWrapper restrictedBeansWrapper) {
...
+	protected void activate(
+		BundleContext bundleContext, Map<String, Object> properties) {
+
 		_freeMarkerEngineConfiguration = ConfigurableUtil.createConfigurable(
 			FreeMarkerEngineConfiguration.class, properties);
 
 		_restrictedVariables = SetUtil.fromArray(
 			_freeMarkerEngineConfiguration.restrictedVariables());
+
+		_serviceTrackerList = ServiceTrackerListFactory.open(
```
Comment (dantewang) [inline modules/apps/portal-template/portal-template-freemarker/src/main/java/com/liferay/portal/template/freemarker/internal/helper/FreeMarkerTemplateContextHelper.java:164]: @W2jx Noticed a new problem recently.

See https://github.com/liferay-core-infra/liferay-portal/pull/2485#pullrequestreview-1189223037 for my explanation.

In short, this `activate` method is also annotated with `@Modified`, so when the configuration of this component changes, the `activate` method will be called again without calling `deactivate`. This means the previous service tracker list will not be closed when you create a new one.

Please move the configuration related logic to a private method, and add a new method to take the `@Modified` annotation. Then you can call the private method from both `activate()` and `modified()`.

File: modules/apps/portal-template/portal-template-velocity/src/main/java/com/liferay/portal/template/velocity/internal/helper/VelocityTemplateContextHelper.java [hunk 2]
```diff
@@ -152,11 +151,21 @@ public void prepare(
 
 	@Activate
 	@Modified
-	protected void activate(Map<String, Object> properties) {
+	protected void activate(
+		BundleContext bundleContext, Map<String, Object> properties) {
+
+		_serviceTrackerList = ServiceTrackerListFactory.open(
```
Comment (dantewang) [inline modules/apps/portal-template/portal-template-velocity/src/main/java/com/liferay/portal/template/velocity/internal/helper/VelocityTemplateContextHelper.java:157]: @W2jx For this one, you can just re-create the configuration object in your modified method.

**Comment by W2jx** [comment]
Hi @dantewang , I have added the method modified(), and reordered commits.

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:sf

**Comment by zhangyan0701** [comment]
No regressions found.

**Comment by W2jx** [comment]
Force-Push is just to resolve conflicts about files `portal-impl/bnd.bnd   portal-kernel/bnd.bnd`

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
@W2jx No regressions found.

**Comment by tinatian** [comment]
@W2jx @dantewang Julius is working to remove TemplateResourceParser, I am going to close this pull, and you do not need to work on this any more.

Thanks.
Tina.

**Pull request closed without merging**